CONTRIBUTING.md

```diff
@@ -44,43 +44,82 @@ Also, if we consider this may damage the stack (modifications in the engine or t
 
-## License
+### Scripting
 
-The stack is under MIT. This choice is based on a wish to allow everyone to use it. Please consider using the same license if you wish to Pull Request some code into the main project.
+1. Scripts should be written in *Python 3.6+*, and as simple as possible. Maximum but balanced comments and verbosity should be considered.
 
-If you do not wish so, but you still wish to contribute, please contact us.
+2. A tool like Ansible is a configuration deployment tool. Ansible job is not to do administration tasks. This is scripts job.
+
+3. Tools, scripts, wrappers, are made to simplify system usage (Shell commands, Ansible, Linux tools, etc.). However, when possible, manual way should always be documented side by side with the automated way, to allow easy debug and simply understand scripts, as the stack also has a teaching objective.
+
+4. If a WebUI is available (and it is planed), manual deployment and usage will remains the default way.
+
+### Variables
+
+1. Variables should be centralized in the inventory as much as possible, to ensure a single location for users to configure all, and non-replication of variables. This is not the Ansible standard, but a need for a unified stack.
+
+2. Internal roles variables, means under the hood variables that user should not consider modifying, can be defined in the role vars. Example: packages names, services names, paths, etc.
+
+3. General variables, i.e. not related to an equipment, can be defined in group_vars/all/general_settings/, in dedicated files or if needed in a new one. When possible, variables should be optional. In any cases, they must be documented in the role readme (and if needed provided commented in the example inventories).
+
+4. All variables related to an equipment should go in group_vars/all/all_equipment/.
 
+5. All variables containing jinja2 code must be prefix with *j2_* and stored in group_vars/all/j2_variables/. j2_ variables are core of the stack, and should be manipulated with care. These are intended to be precedence by user only. If needed, theses variables can be fixed.
 
+6. Also, consider that some variables are specific in BlueBanquise, and do not fully follow the standard rules of Ansible: *Equipment_profile* and *Authentication* dictionaries are good example.
+These dictionaries **must** be set in group_vars/equipment_profile/ folders or in group_vars/all/. Any other usage (for example at host_vars level or extra_vars level) will result in a unpredictable situation.
 
+### Miscellaneous
 
+1. We start numbering at **1**, since a cluster stack is related to existing physical elements. Networks start at 1, icebergs at 1, etc.
 
+2. We are using *.yml* (and not *.yaml*) for all YAML files.
+
+3. We try to not use tabulations, but instead double space for indentation.
+
+4. If you are working on Microsoft Windows, please check you do not submit ^M at the end of your lines.
```

Comment:
That's called CRLF, displayed as ^M with Linux systems. Maybe add a link to git documentation https://www.git-scm.com/book/en/v2/Customizing-Git-Git-Configuration#_code_core_autocrlf_code

Reply:
From @strus38 : Note: on windows10 WSL running Ubuntu 18.04, you must define in your Ubuntu bash:
`git config --global core.autocrlf `input to prevent LF/CRLF issues.

Reply:
Ok, I will correct this and call it CRLF.
On windows 10, when installing git, installer propose you to ensure no CRLF are commited.